python/ray/data/dataset.py

```diff
@@ -1380,12 +1386,24 @@ def repartition(
         Returns:
             The repartitioned :class:`Dataset`.
         """  # noqa: E501
+
```

Comment:
Is `max_num_rows_per_block` mutually exclusive with `num_blocks`? If so, let's raise a `ValueError` if you specify both, and also describe that they're mutually exclusive in the argument descriptions?